src/olympia/lib/es/tests/test_commands.py

```diff
@@ -121,20 +116,22 @@ def run(self):
             connection._commit()
             connection.clean_savepoints()
 
-        # We should still be able to search in the foreground while the reindex
-        # is being done in the background. We should also be able to index new
-        # documents, and they should not be lost.
-        old_addons_count = len(self.expected)
-        while t.is_alive() and len(self.expected) < old_addons_count + 3:
-            self.expected.append(addon_factory())
-            connection._commit()
-            connection.clean_savepoints()
-            self.refresh()
-            self.check_results(self.expected)
-
-        if len(self.expected) == old_addons_count:
-            raise AssertionError('Could not index objects in foreground while '
-                                 'reindexing in the background.')
+        if not wipe:
+            # We should still be able to search in the foreground while the
+            # reindex is being done in the background. We should also be able
+            # to index new documents, and they should not be lost.
+            old_addons_count = len(self.expected)
+            while t.is_alive() and len(self.expected) < old_addons_count + 3:
+                self.expected.append(addon_factory())
+                connection._commit()
+                connection.clean_savepoints()
+                self.refresh()
+                self.check_results(self.expected)
+
+            if len(self.expected) == old_addons_count:
+                raise AssertionError(
+                    'Could not index objects in foreground while reindexing '
+                    'in the background.')
```

Comment:
So... what happens if you try to index a new document at the same time as a new index is being built when the old one is being wiped?  Should that work?  (or is it something we never do on live systems because it doesn't)

Reply:
It would try to create the index (which doesn't exist yet) automatically, which would leave a mess - I think creating the alias at the end would then fail. It should never be used in production with a live cluster unless absolutely necessary, it's more of an initial setup thing.